chia/full_node/mempool.py

```diff
@@ -90,66 +172,76 @@ def remove_from_pool(self, items: List[bytes32], reason: MempoolRemoveReason) ->
         """
         Removes an item from the mempool.
         """
-        for spend_bundle_id in items:
-            item: Optional[MempoolItem] = self._spends.get(spend_bundle_id)
-            if item is None:
-                continue
-            assert item.name == spend_bundle_id
-            removals: List[Coin] = item.removals
-            for rem in removals:
-                rem_name: bytes32 = rem.name()
-                self._removal_coin_id_to_spendbundle_ids[rem_name].remove(spend_bundle_id)
-                if len(self._removal_coin_id_to_spendbundle_ids[rem_name]) == 0:
-                    del self._removal_coin_id_to_spendbundle_ids[rem_name]
-            del self._spends[item.name]
-            del self._sorted_spends[item.fee_per_cost][item.name]
-            dic = self._sorted_spends[item.fee_per_cost]
-            if len(dic.values()) == 0:
-                del self._sorted_spends[item.fee_per_cost]
-            self._total_mempool_cost = CLVMCost(uint64(self._total_mempool_cost - item.cost))
-            self._total_mempool_fees = self._total_mempool_fees - item.fee
-            assert self._total_mempool_cost >= 0
-            info = FeeMempoolInfo(self.mempool_info, self._total_mempool_cost, self._total_mempool_fees, datetime.now())
-            if reason != MempoolRemoveReason.BLOCK_INCLUSION:
-                self.fee_estimator.remove_mempool_item(
-                    info, MempoolItemInfo(item.cost, item.fee, item.height_added_to_mempool)
-                )
+        if items == []:
+            return
+
+        removed_items: List[MempoolItemInfo] = []
+        if reason != MempoolRemoveReason.BLOCK_INCLUSION:
+
+            for spend_bundle_ids in chunks(items, SQLITE_MAX_VARIABLE_NUMBER):
+                args = ",".join(["?"] * len(spend_bundle_ids))
+                with self._db_conn:
+                    cursor = self._db_conn.execute(
+                        f"SELECT name, cost, fee FROM tx WHERE name in ({args})", spend_bundle_ids
```

Comment:
Arvid points out that `remove_from_pool` can now throw `KeyError`.
The important call sites are:
mempool_manager.py:new_peak
mempool_manager.py:add_spend_bundle
mempool.py:add_to_pool

Let's go over those in the meeting.

Reply:
`add_to_pool`: this call site seems safe because if we throw from here, we will not add a tx we should not have.
`add_spend_bundle`: seems safe, because we will throw before `add_to_pool`

`new_peak`: This one may not be OK. We can throw in the middle of a loop that is removing spends from the mempool. It seems that removing fewer spends than intended in this case is safe, but may prevent new spends being added that otherwise would have been if all the other spends had been removed.

Reply:
you mean the `new_peak()` function in `mempool_manager.py`, right? I'm confident that's a place where we won't throw, since we only collect transactions that already exist in the mempool, and then remove them.